addons/l10n_ch/models/account_invoice.py

```diff
@@ -43,228 +25,38 @@ def _compute_l10n_ch_qr_is_valid(self):
-            (3) 4 | control digit for amount
-            (4) 150001 | id number of the customer (size may vary, usually 6 chars)
-            (5) 12345678901234567890 | reference
-            (6) 1: control digit for identification number and reference
-            (7) 010234567: subscription number (01-23456-7)
-        """
-        for record in self:
-            record.l10n_ch_isr_optical_line = ''
-            if record.l10n_ch_isr_number and record.l10n_ch_isr_subscription and record.currency_id.name:
-                # Final assembly (the space after the '+' is no typo, it stands in the specs.)
-                record.l10n_ch_isr_optical_line = '{amount}>{reference}+ {creditor}>'.format(
-                    amount=record._get_l10n_ch_isr_optical_amount(),
-                    reference=record.l10n_ch_isr_number,
-                    creditor=record.l10n_ch_isr_subscription,
-                )
-
-    @api.depends(
-        'move_type', 'name', 'currency_id.name',
-        'partner_bank_id.l10n_ch_isr_subscription_eur',
-        'partner_bank_id.l10n_ch_isr_subscription_chf')
-    def _compute_l10n_ch_isr_valid(self):
-        """Returns True if all the data required to generate the ISR are present"""
-        for record in self:
-            record.l10n_ch_isr_valid = record.move_type == 'out_invoice' and\
-                record.name and \
-                record.l10n_ch_isr_subscription and \
-                record.l10n_ch_currency_name in ['EUR', 'CHF']
+        if self.partner_bank_id.l10n_ch_qr_iban and self.l10n_ch_is_qr_valid and self.name:
+            invoice_ref = re.sub(r'[^\d]', '', self.name)
+            # keep only the last digits if it exceed boundaries
+            ref_payload_len = l10n_ch_QRR_NUMBER_LENGTH - 1
+            extra = len(invoice_ref) - ref_payload_len
+            if extra > 0:
+                invoice_ref = invoice_ref[extra:]
+            internal_ref = invoice_ref.zfill(ref_payload_len)
+            return mod10r(internal_ref)
+        else:
+            return False
 
     @api.depends('move_type', 'partner_bank_id', 'payment_reference')
-    def _compute_l10n_ch_isr_needs_fixing(self):
+    def _compute_l10n_ch_qrr_needs_fixing(self):
         for inv in self:
             if inv.move_type == 'in_invoice' and inv.company_id.account_fiscal_country_id.code in ('CH', 'LI'):
                 partner_bank = inv.partner_bank_id
-                needs_isr_ref = partner_bank.l10n_ch_qr_iban or partner_bank._is_isr_issuer()
-                if needs_isr_ref and not inv._has_isr_ref():
-                    inv.l10n_ch_isr_needs_fixing = True
+                if partner_bank.l10n_ch_qr_iban and not inv._l10n_ch_has_valid_ref():
+                    inv.l10n_ch_qrr_needs_fixing = True
                     continue
-            inv.l10n_ch_isr_needs_fixing = False
+            inv.l10n_ch_qrr_needs_fixing = False
 
-    def _has_isr_ref(self):
+    def _l10n_ch_has_valid_ref(self):
```

Comment:
```suggestion
    def _l10n_ch_has_valid_qrr(self):
```
Wouldn't this be more accurate ?

Reply:
I actually can't find a situation not using the ISR bill which would use this function thinking